File: jokercsi/Programmers/전화번호 목록.py
def solution(phone_book):
    answer = True

    # phone book의 모든 번호 중 하나가 다른 번호가 접두어면 false
    dict = {}
    phone_book.sort()
    
    # key값만 있는 해시 생성
    for i in range(len(phone_book)):
        dict[phone_book[i]] = 1 # value는 1으로 통합
    
    for number in phone_book:
        makeword = ''
        for num in number:
            makeword += num
            if makeword in dict and number != makeword:
                # print(number, makeword)
                answer = False
                break
    
    return answer

File: ise-yen/PGS/전화번호 목록.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool solution(vector<string> phone_book) {
    bool answer = true;
    if(phone_book.size() == 1) return answer;
    
    sort(phone_book.begin(), phone_book.end());
    
    string flag = phone_book[0];
    for(int i = 1; i < phone_book.size(); i++){
        string pb = phone_book[i];
        if(flag.length() <= pb.length() && flag == pb.substr(0, flag.length())) return false;
        else flag = pb;
    }
    
    return answer;
}
